packages/eslint-plugin/docs/rules/consistent-type-exports.md

```diff
@@ -70,15 +70,14 @@ export type { ButtonProps } from 'some-library';
 ### ✅ Correct
 
 ```ts
-export { Button } from 'some-library';
-export type { ButtonProps } from 'some-library';
 export { Button, type ButtonProps } from 'some-library';
 ```
 
 ## When Not To Use It
 
 - If you are using a TypeScript version less than 3.8, then you will not be able to use this rule as type exports are not supported.
 - If you specifically want to use both export kinds for stylistic reasons, you can disable this rule.
+- Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.
```

Comment:
Probably a minor rewording so it goes more along the lines of "If you use `--isolatedModules` you don't need this rule since TS would enforce type-export specifiers"